components/src/hooks/useSelectDeckLocation/index.tsx

```diff
@@ -103,11 +141,13 @@ export function DeckLocationSelect({
             addressableArea.id === 'fixedTrash'
         )
         .map(slot => {
+          // eslint-disable-next-line react-hooks/rules-of-hooks
           const slotLocation = { slotName: slot.id }
-          const isDisabled = disabledLocations.some(
-            l =>
-              typeof l === 'object' && 'slotName' in l && l.slotName === slot.id
-          )
+          const isDisabled = !availableSlotNames.some(l => l === slot.id)
```

Comment:
nit: can the l be more descriptive? maybe something like `slotName`? 

```suggestion
          const isDisabled = !availableSlotNames.some(slotName => slotName === slot.id)
```